fw/http.c

```diff
@@ -6501,6 +6532,9 @@ tfw_http_resp_process(TfwConn *conn, TfwStream *stream, struct sk_buff *skb)
 	 * connection and its corresponding freeing.
 	 */
 	if (websocket) {
+		/* Don't upgrade if client connection has been closed. */
+		if (test_bit(TFW_HTTP_B_REQ_DROP, hmresp->req->flags))
+			return TFW_BAD;
```

Comment:
It seems a client can close its connection right after the check and before the `tfw_http_websocket_upgrade()` at the below, so the check seems unreliable. What happens if we just remove the check and upgrade the connection? Won't it be closed on the response forwarding?

The idea is to free all resources on a connection closing event through reference counting. In my understanding we won't find the client connection freed thanks to reference counting and we can just ignore the check...

Reply:
Then shouldn't we remove it? Or maybe I didn't understand you...

Reply:
Yes, i have to remove it.